pkg/resource/deploy/source_eval.go

```diff
@@ -300,7 +300,7 @@ func (d *defaultProviders) newRegisterDefaultProviderEvent(
 	event := &registerResourceEvent{
 		goal: resource.NewGoal(
 			providers.MakeProviderType(req.Package()),
-			req.Name(), true, inputs, "", false, nil, "", nil, nil, nil, nil, nil, nil, "", nil),
+			req.Name(), true, inputs, "", false, nil, "", nil, nil, nil, nil, nil, nil, "", nil, false),
```

Comment:
Would it be worth wrapping these params in a struct? This list is getting unwieldy.

Reply:
I was thinking about that, too, esp. given that so many are essentially optional. Ironically, all this method does is... build a struct. So we could just use it directly and get rid of these incomprehensible calls.